api-report/merge-tree.api.md

```diff
@@ -590,6 +594,9 @@ export interface ISegment extends IMergeNodeCommon, Partial<IRemovalInfo> {
     clientId: number;
     // (undocumented)
     clone(): ISegment;
+    readonly isEndEndpoint?: boolean;
```

Comment:
Aim to make your API align with semantics. There are only 3 possibilities here: 'not an endpoint', 'start', or 'end'. I'd prefer something like `readonly endpointType: 'start' | 'end' | undefined` (maybe making it optional rather than explicit undefined).